File: src/types/substitution.ts
export interface SubstitutionRecord {
  id: number;
  originalEntryId: number;
  substituteTeacherId: number;
  date: string;
  decisionReason: string;
  score: number;
  createdAt: string;
}

export interface NewSubstitutionRecord {
  originalEntryId: number;
  substituteTeacherId: number;
  date: string;
  decisionReason?: string;
  score?: number;
}

export interface SubstitutionCandidate {
  teacherId: number;
  teacherName: string;
  score: number;
  scoreBreakdown: ScoreBreakdown;
  decisionReason: string;
  isQualified: boolean;
}

export interface ScoreBreakdown {
  engagement: number;
  substitutionLoad: number;
  pedagogical: number;
  weeklyLoad: number;
  subjectQualification: number;
}

export interface AffectedEntry {
  entryId: number;
  timeSlotId: number;
  period: number;
  startTime: string;
  endTime: string;
  classId: number;
  className: string;
  subjectId: number;
  subjectName: string;
  originalTeacherId: number;
  originalTeacherName: string;
  roomId: number;
  roomName: string;
  isSubstituted: boolean;
  substituteTeacherName: string | null;
}

File: src-tauri/src/models/substitution.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SubstitutionRecord {
    pub id: i64,
    pub original_entry_id: i64,
    pub substitute_teacher_id: i64,
    pub date: String,
    pub decision_reason: String,
    pub score: f64,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct NewSubstitutionRecord {
    pub original_entry_id: i64,
    pub substitute_teacher_id: i64,
    pub date: String,
    pub decision_reason: Option<String>,
    pub score: Option<f64>,
}
